### PR TITLE
Remove IPv4 from docs; update how to retrieve IPv6 address

### PR DESCRIPTION
**Pull request by jheidbrink**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -98,19 +98,17 @@ but usually we will act ASAP anyway.
 
 after you have been approved..
 
-check if your pubkey file is in the current master of https://github.com/krebs/retiolum
-look into your pubkey file and copy your ipv6 address which got autogenerated.
+To learn your IP address, run `nix eval --impure --expr '((import ./lib).krebs.genipv6 "retiolum" "<your_username>" { hostName = "<your_hostname>"; }).address'` in the stockholm repo.
```
Comment (4z3) [inline README.md]: This is actually not necessarily true. `genipv6` is an optional convenience function, even though everybody uses it (I think.)  The only structure that's mandatory is `42:${network-id}:${hash username}::/48`, the other bits are user-specified.

Comment (4z3) [reply README.md]: So I'd recommend to retrieve the actually configured address.

File: README.md [hunk 2]
```diff
@@ -98,19 +98,17 @@ but usually we will act ASAP anyway.
...
 copy the `example.nix` to your configuration.nix, update the rev and sha256 of the retiolum fetchgit to the current master. (you have to update it everytime if new hosts join you want to connect to)
 import the example.nix (you can choose a better name, like retiolum.nix) in your configuration.nix
-configure your ipv4 and ipv6 addresses
+configure your ipv6 address
```
Comment (4z3) [inline README.md:105]: :+1:

File: example.nix [hunk 3]
```diff
@@ -15,12 +15,6 @@ let
 
 in {
   options = {
-    networking.retiolum.ipv4 = mkOption {
-      type = types.str;
-      description = ''
-        own ipv4 address
-      '';
-    };
```
Comment (4z3) [inline example.nix:23]: Hm, I think we do want to keep it optional. `nullOr str` with a default of `null` maybe?

File: example.nix [hunk 4]
```diff
@@ -66,9 +67,10 @@ in {
         Name = tinc.${netname}
 
         [Network]
-        Address=${cfg.ipv4}/12
         Address=${cfg.ipv6}/16
-      '';
+      '' + (if cfg.ipv4 != null then ''
+        Address=${cfg.ipv4}/12
+      '' else "");
```
Comment (4z3) [inline example.nix]: `+ optionalString (cfg.ipv4 != null) ...`

File: example.nix (hunk 3 above)
Comment (jheidbrink) [reply example.nix:23]: I force-pushed an attempt to do that. Tested by adding and removing the `networking.retiolum.ipv4 =` line and verified that the device is configured correctly.

**Comment by 4z3** [comment]
Thanks.

**Pull request merged**